apps/desktop/desktop/src/main/java/bisq/desktop/main/content/settings/bisq_connect/BisqConnectController.java

```diff
@@ -0,0 +1,252 @@
+                        model.getQrPlaceholder().set(Res.get("settings.bisqConnect.connectionDetails.placeholderLan"));
+                    } else {
+                        model.getQrPlaceholder().set(Res.get("settings.bisqConnect.connectionDetails.placeholderTor"));
+                    }
+                }
+            }
+        } catch (Exception e) {
+            model.getApiUrl().set("");
+            model.getQrCodeImage().set(null);
+            model.getQrPlaceholder().set(Res.get("settings.bisqConnect.connectionDetails.placeholderLan.error"));
+        }
+    }
+
+    private void applyUrlToModel(String url) {
+        model.getApiUrl().set(url);
+        model.getQrPlaceholder().set("");
+        try {
+            model.getQrCodeImage().set(QrCodeDisplay.toImage(url, model.getQrCodeSize().get()));
+        } catch (RuntimeException e) {
+            log.warn("QR code generation failed", e);
+            model.getQrCodeImage().set(null);
+            model.getQrPlaceholder().set(Res.get("settings.bisqConnect.connectionDetails.placeholderError"));
+        }
+    }
+
+    private void writeCustomConfig() {
+        ApplicationService.Config appConfig = serviceProvider.getConfig();
+        Path appDataDirPath = appConfig.getAppDataDirPath();
+
+        boolean torMode = model.getSelectedMode().get().isTor();
+        String password = Optional.ofNullable(model.getPassword().get()).orElse("");
+
+        Config newConfig = ConfigFactory.parseMap(Map.of(
+                "application.websocket.enabled", model.getEnabled().get(),
+                "application.websocket.localhostOnly", torMode,
+                "application.websocket.publishOnionService", torMode,
+                "application.websocket.server.host", torMode ? "localhost" : "0.0.0.0",
+                "application.websocket.password", password
+        ));
+
+        Config customConfig = TypesafeConfigUtils.resolveCustomConfig(appDataDirPath).orElse(ConfigFactory.empty());
+        Config config = newConfig.withFallback(customConfig).resolve();
+
+        String rendered = config.root().render(ConfigRenderOptions.defaults()
+                .setOriginComments(false)
+                .setJson(false)
+                .setFormatted(true));
+
+        Path customConfigFilePath = appDataDirPath.resolve(ApplicationService.CUSTOM_CONFIG_FILE_NAME);
+        try {
+            FileMutatorUtils.writeToPath(rendered, customConfigFilePath);
+        } catch (IOException e) {
+            log.error("Could not write config file {}", customConfigFilePath.toAbsolutePath(), e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void updateChangeDetectedFlag() {
+        boolean changeDetected = model.getEnabled().get() != savedEnabled
+                || model.getSelectedMode().get() != savedMode
+                || !Objects.equals(model.getPassword().get(), savedPassword);
+        UIThread.run(() -> model.getIsChangeDetected().set(changeDetected));
```

Comment:
UIThread.run is not needed here as that method is only called from UI handlers. We only need UIThread.run when we get called from domain layer (non Java FX).